compiler/optimizations/gpuTransforms.cpp

```diff
@@ -635,7 +635,7 @@ void GpuKernel::populateBody(CForLoop *loop, FnSymbol *outlinedFunction) {
       DefExpr* newDef = def->copy();
       this->copyMap_.put(def->sym, newDef->sym);
 
-      outlinedFunction->insertAtTail(newDef);
+      outlinedFunction->insertBeforeEpilogue(newDef);
```

Comment:
OOC: what required this change, do you remember?

Reply:
The AST for the outlined function looks something like this:

```
function chpl_gpu_kernel(...)
{
  ... some upfront logic about calculating index ...
  if(idx > loopUpperBound)
  {
      gotoReturn _end_chpl_gpu_kernel[2002168] _end_chpl_gpu_kernel[2002168]
  }

  ... body of loop we're outlining ...

  label _end_chpl_gpu_kernel[2002168]
  (2002170 return _void[58])
  (2002074 'local block')
}
```

When we're populating that "body of loop we're outlining" section we don't want to dump it into the epilogue that'll get executed for out-of-bound indices so I call `insertBeforeEpilogue` instead.